Tests/TuistKitTests/Commands/GraphCommandTests.swift

```diff
@@ -0,0 +1,65 @@
+import Basic
+import Foundation
+import SPMUtility
+import TuistCore
+import XcodeProj
+import XCTest
+@testable import TuistCoreTesting
+@testable import TuistKit
+
+final class GraphCommandTests: XCTestCase {
+    var subject: GraphCommand!
+    var dotGraphGenerator: MockDotGraphGenerator!
+    var fileHandler: MockFileHandler!
+    var printer: MockPrinter!
+    var manifestLoader: MockGraphManifestLoader!
+    var parser: ArgumentParser!
+
+    override func setUp() {
+        super.setUp()
+        dotGraphGenerator = MockDotGraphGenerator()
+        fileHandler = try! MockFileHandler()
+        printer = MockPrinter()
+        manifestLoader = MockGraphManifestLoader()
+        parser = ArgumentParser.test()
+
+        subject = GraphCommand(parser: parser,
+                               fileHandler: fileHandler,
+                               printer: printer,
+                               dotGraphGenerator: dotGraphGenerator,
+                               manifestLoader: manifestLoader)
+    }
+
+    func test_command() {
+        XCTAssertEqual(GraphCommand.command, "graph")
+    }
+
+    func test_overview() {
+        XCTAssertEqual(GraphCommand.overview, "Generates a dot graph from the workspace or project in the current directory.")
+    }
+
+    func test_run() throws {
+        // Given
+        let graphPath = fileHandler.currentPath.appending(component: "graph.dot")
+        let projectManifestPath = fileHandler.currentPath.appending(component: "Project.swift")
+
+        try fileHandler.touch(graphPath)
+        try fileHandler.touch(projectManifestPath)
+
+        manifestLoader.manifestsAtStub = {
+            if $0 == self.fileHandler.currentPath { return Set([.project]) }
+            else { return Set([]) }
+        }
+
+        let graph = "graph {}"
+        dotGraphGenerator.generateProjectStub = graph
+
+        // When
+        let result = try parser.parse([GraphCommand.command])
+        try subject.run(with: result)
+
+        // Then
+        XCTAssertTrue(printer.printArgs.contains("Deleting existing graph at \(graphPath.pathString)"))
+        XCTAssertTrue(printer.printSuccessArgs.contains("Graph exported to \(graphPath.pathString)"))
```

Comment:
is it worth validating the generate dot file contains `"graph {}"`? (not sure myself, I guess this test is mostly for the command line input / output)

Reply:
That's a good idea. I updated the test with that suggestion.